### ISSUE TITLE
Stream.chunk/4 broken on 1.5.0-rc.1

### ISSUE THREAD

**Issue by wojtekmach** [elixir-lang/elixir#6319]
### Precheck

* Elixir & Erlang versions (elixir --version): v1.5.0-rc.1
* Operating system: macOS Sierra

### Current behavior

```elixir
iex(1)> {:ok, pid} = StringIO.open("foo\nbar\nbaz") ; IO.stream(pid, :line) |> Stream.chunk(1, 1, []) |> Enum.to_list
** (UndefinedFunctionError) function Stream.Reducers.chunk/5 is undefined or private. Did you mean one of:

      * chunk_by/3
      * chunk_every/5
      * chunk_while/1
      * chunk_while/2

    (elixir) Stream.Reducers.chunk(#Function<1.117392430/4 in Stream.chunk/4>, %IO.Stream{device: #PID<0.654.0>, line_or_bytes: :line, raw: false}, 1, 1, [])
```

### Expected behavior

behaviour from v1.5.0-rc.0:

```elixir
iex(5)> {:ok, pid} = StringIO.open("foo\nbar\nbaz") ; IO.stream(pid, :line) |> Stream.chunk(1, 1, []) |> Enum.to_list
[["foo\n"], ["bar\n"], ["baz"]]
```

**Comment by gmile** [comment]
Seeing the same as of Elixir 1.6.0-dev (de365cd).

**Comment by josevalim** [comment]
Fixed on master and v1.5 branches.

**Comment by josevalim** [comment]
Thanks for reporting! 
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
